pallets/liquid-staking/src/lib.rs

```diff
@@ -390,16 +398,37 @@ pub mod pallet {
             T::WithdrawOrigin::ensure_origin(origin)?;
             ensure!(T::Members::contains(&agent), Error::<T>::IllegalAgent);
 
-            TotalStakingAsset::<T>::try_mutate(|b| -> DispatchResult {
-                *b = b.checked_sub(amount).ok_or(ArithmeticError::Underflow)?;
-                Ok(())
-            })?;
-            let exchange_rate = Rate::checked_from_rational(
-                TotalStakingAsset::<T>::get(),
-                TotalVoucher::<T>::get(),
-            )
-            .ok_or(Error::<T>::InvalidExchangeRate)?;
-            ExchangeRate::<T>::put(exchange_rate);
+            let reserves = Self::total_reserves();
+            let left_reserves = reserves.saturating_sub(amount);
+            let reduced_reserves = reserves
```

Comment:
This function about insurance pool is confused.
The changes in my mind is:
* create a pool, and deposits insurance fund from specific origin
* liquid-staking rely on an insurance provider (trait), when slash happens, use this insurance provider, the provider should be able to return the covered slashed funds, and the uncovered funds. If there are uncovered slash funds, then remove total staking asset and update exchange rate.
* (nice to have) the insurance funds can deposits in lending market and get interest.

Reply:
@kaichaosun the second point is actually covered already, we do calculate the covered slashes, but yes, we didn't allow any users to deposit funds.